revocation/ocsp/errors.go

```diff
@@ -0,0 +1,58 @@
+// Package OCSP provides methods for checking the OCSP revocation status of a certificate chain
+// as well as errors related to these checks
+package ocsp
+
+import (
+	"fmt"
+	"time"
+)
+
+// RevokedError is returned when the certificate's status for OCSP is ocsp.Revoked
+type RevokedError struct{}
+
+func (e RevokedError) Error() string {
+	return "certificate is revoked via OCSP"
+}
+
+// UnknownStatusError is returned when the certificate's status for OCSP is ocsp.Unknown
+type UnknownStatusError struct{}
+
+func (e UnknownStatusError) Error() string {
+	return "certificate has unknown status via OCSP"
+}
+
+// CheckOCSPError is returned when there is an error during the OCSP revocation check, not necessarily a revocation
+type CheckOCSPError struct {
+	Err error
+}
+
+func (e CheckOCSPError) Error() string {
+	msg := "error checking revocation status via OCSP"
+	if e.Err != nil {
+		return fmt.Sprintf("%s: %s", msg, e.Err.Error())
+	}
+	return msg
+}
+
+// NoServerError is returned when the OCSPServer is not specified or is not an HTTP URL.
+type NoOCSPServerError struct{}
+
+func (e NoOCSPServerError) Error() string {
+	return "no valid OCSPServer found"
+}
+
+// PKIXNoCheckError is returned when the OCSP signing cert is missing the id-pkix-ocsp-nocheck extension.
+type PKIXNoCheckError struct{}
+
+func (e PKIXNoCheckError) Error() string {
+	return fmt.Sprintf("an OCSP signing cert is missing the id-pkix-ocsp-nocheck extension (%s)", pkixNoCheckOID)
+}
+
+// TimeoutError is returned when the connection attempt to an OCSP URL exceeds the specified threshold
+type TimeoutError struct {
+	timeout time.Duration
+}
+
+func (e TimeoutError) Error() string {
+	return fmt.Sprintf("exceeded timeout threshold of %.2f seconds for OCSP check", e.timeout.Seconds())
+}
```

Comment:
Does timeout error falls into `CheckOCSPError` bucket?

Reply:
No, I separated these for more granularity. Since a timeout is more flexible (i.e. it can change between responses, and can sometimes be resolved by adjusting the timeout threshold), I decided to separate it from CheckOCSPError, which captures any other errors that are encountered during the process. I figure that CheckOCSPError means that there is an actual issue with the request/response/chain/cert that needs to be addressed before a proper status can be determined.

Reply:
@priteshbandi does this answer your question? If so, I can resolve this